contracts/governance/FeeSharingCollector/FeeSharingCollectorMultipleToken.sol

```diff
@@ -0,0 +1,622 @@
+     * For SOV, it will be directly deposited into the FeeSharingCollectorMultipleToken from the protocol.
+     *
+     * @param _tokens array address of the token
+     * */
+    function withdrawFees(address[] memory _tokens) public {
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            require(
+                Address.isContract(_tokens[i]),
+                "FeeSharingCollectorMultipleToken::withdrawFees: token is not a contract"
+            );
+        }
+
+        uint256 wrbtcAmountWithdrawn = protocol.withdrawFees(_tokens, address(this));
+        uint256 poolTokenAmount;
+
+        address wRBTCAddress = wrappedNativeTokenAddress;
+        require(
+            wRBTCAddress != address(0),
+            "FeeSharingCollectorMultipleToken::withdrawFees: wRBTCAddress is not set"
+        );
+
+        address loanPoolToken = protocol.underlyingToLoanPool(wRBTCAddress);
+        require(
+            loanPoolToken != address(0),
+            "FeeSharingCollectorMultipleToken::withdrawFees: loan wRBTC not found"
+        );
+
+        if (wrbtcAmountWithdrawn > 0) {
+            /// @dev TODO can be also used - function addLiquidity(IERC20Token _reserveToken, uint256 _amount, uint256 _minReturn)
+            IERC20(wRBTCAddress).approve(loanPoolToken, wrbtcAmountWithdrawn);
+            poolTokenAmount = ILoanToken(loanPoolToken).mint(address(this), wrbtcAmountWithdrawn);
+
+            /// @notice Update unprocessed amount of tokens
+            uint96 amount96 = safe96(
+                poolTokenAmount,
+                "FeeSharingCollectorMultipleToken::withdrawFees: pool token amount exceeds 96 bits"
+            );
+
+            _addCheckpoint(loanPoolToken, amount96);
+        }
+
+        emit FeeWithdrawn(msg.sender, loanPoolToken, poolTokenAmount);
+    }
+
+    /**
+     * @notice Withdraw fees from sovryn dex:
+     * protocolFee from the conversion
+     * the fees will be converted in wrappedNativeToken form, and then will be transferred to wrappedNativeToken loan pool
+     *
+     * @param _tokens array addresses of the tokens
+     * */
+    function withdrawFeesFromDex(address[] memory _tokens) public {
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            require(
+                Address.isContract(_tokens[i]),
+                "FeeSharingCollector::withdrawFeesFromDex: token is not a contract"
+            );
+        }
+
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            address token = _tokens[i];
+            /** Withdraw from dex */
+            bytes memory cmd = abi.encode(SOVRYN_DEX_CMD_COLLECT_TREASURY_CODE, token);
+            ISovrynDex(sovrynDexAddress).userCmd(SOVRYN_DEX_COLD_PATH_PROXY_IDX, cmd);
+        }
+    }
+
+    /**
+     * @notice Transfer tokens to this contract.
+     * @dev We just update amount of tokens here and write checkpoint in a separate methods
```

Comment:
pls add a @dev note that the caller should take care of setting allowance for the token

Reply:
addressed